torchao/csrc/metal/int4_quantized_kernels.metal

```diff
@@ -0,0 +1,128 @@
+#include <metal_simdgroup>
```

Comment:
separating bit widths by file seems like it will preclude us from code sharing; MLX has shown that we can probably share significant portions of 2-bit and 4-bit kernels (and, if anyone wants it, groupwise-quantized 8-bit), right?

Reply:
I didnt intend to make this exclusive for int4. Way MLX handles this makes sense, so when we add 2 bit support, it will definitely look like that